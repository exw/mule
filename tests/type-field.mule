{
  , type t = int
  , x : t = 4
}
