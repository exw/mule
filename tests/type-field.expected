
Desugared:

	(App
	 ((Fix Record)
	  (Lam
	   ($0
	    (LetType
	     (((t (Recur (() t (Var (() int))))))
	      (App
	       ((App
	         ((Update (Type t))
	          (App
	           ((App ((Update (Value x)) EmptyRecord))
	            (App ((WithType (Var (() t))) (Integer 4)))))))
	        (Witness (Var (() t)))))))))))

inferred type:

	(Quant
	 (() All a
	  (Record (r_info ()) (r_types (() ((t (Named (() int)))) (a)))
	   (r_values (() ((x (Named (() int)))) ())))))

Runtime term:

	(App ((Fix Record) (Lam (0 () (Record ((x (Integer 4))))))))

Evaluated:

	(Record ((x (Integer 4))))
